packages/kas/src/nodes.ts

```diff
@@ -1976,13 +2026,17 @@ export class Pow extends Expr {
 
 /* logarithm */
 export class Log extends Expr {
-    constructor(base, power) {
+    base: Expr;
+    power: Expr;
+
+    constructor(base: Expr, power: Expr) {
         super();
         this.base = base;
         this.power = power;
-        this.hints = _.extend(this.hints, {
+        this.hints = {
+            ...this.hints,
```

Comment:
These are hints from the parent class?

Reply:
That's correct.  I believe that `Expr` sets `parens: false` or something like that.